qos/judge/state.go

```diff
@@ -0,0 +1,86 @@
+package judge
+
+import (
+	"sync"
+
+	"github.com/pokt-network/poktroll/pkg/polylog"
+)
+
+// serviceState maintains the state for a QoS service.
+// It provides methods for:
+// - Updating state parameters using observations.
+// - Updating endpoint results using observations
+// - Reading state parameters and endpoints (Read Only) for:
+//   - building endpoint results
+//   - endpoint selection
+type ServiceState struct {
+	// logger for diagnostics
+	logger polylog.Logger
+
+	// mu protects the state map from concurrent access
+	mu sync.RWMutex
+
+	// stateParameters
+	parameters map[string]*StateParameter
+
+	// endpoint store maintained by the service state.
+	// declared embedded to allow direct access to the store methods, e.g. getEndpoint
+	*endpointStore
+}
+
+func (s *ServiceState) GetStrParam(paramName string) (string, bool) {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	param, ok := s.parameters[paramName]
+	if !ok {
+		return "", false
+	}
+
+	return param.GetStr()
+}
+
+func (s *ServiceState) GetIntParam(paramName string) (int, bool) {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	param, ok := s.parameters[paramName]
+	if !ok {
+		return 0, false
+	}
+
+	return param.GetInt()
+
+}
+
+func (s *ServiceState) GetConsensusParam(paramName string) (map[string]int, bool) {
```

Comment:
#PUC for this since it's a bit more complicated that the simple values. Specifically what it's used for (eg. archival checks).